### ISSUE TITLE
Screen Reader issue for Multi SelectMenu

### ISSUE THREAD

**Issue by Lakshya-Poddar** [tailwindlabs/headlessui#2664]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.7.16

**What browser are you using?**

Chrome, Safari

**Reproduction URL**

A public GitHub repo that includes a minimal reproduction of the bug. **Please do not link to your actual project**, what we need instead is a _minimal_ reproduction in a fresh project without any unnecessary code. This means it doesn't matter if your real project is private/confidential, since we want a link to a separate, isolated reproduction anyways. Unfortunately we can't provide support without a reproduction, and your issue will be closed with no comment if this is not provided.

You can use one of the starting projects on CodeSandbox:

CodeSandbox Link: https://codesandbox.io/s/hungry-wescoff-jr98dt

**Describe your issue**

In the Multi SelectMenu dropdown, on hovering or navigating the dropdown items using arrow keys the screen reader reads it as selected/not-selected, selected (<index of the item>/<total items>). But ideally it should announce something like selected/not-selected, focused (<index of the item>/<total items>).

**Comment by RobinMalfait** [comment]
Hey!

Unfortunately we don't have control over that. We are currently setting the `aria-multiselectable` attribute, and each selected item receives a `aria-selected` attribute. 

However, when you are navigating through the list in Safari you will hear/see this:
- For non-selected items:
  <img width="646" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/7f924eb8-18df-46b9-b3c4-8853488fbac5">
- For selected items:
  <img width="646" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/64532d66-1a33-4d91-b41f-3907cd8b7df8">

For Chrome this looks like:
- For non-selected items:
  <img width="646" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/6d8356a3-433c-4c22-a9c4-0d5458f0ee97">
- For selected items:
  <img width="646" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/1834413/8063d7bf-1823-48f8-8979-00966f6a8b95">

So we don't have control over that exact message that is being displayed / read out loud. This is done by the browser itself.

Luckily there can only ever be 1 item in the "active" state (via hover or using keyboard navigation), so the currently read out value is always the "active" one.

So while I agree that ideally we have a bit more control over that and make it more consistent, unfortunately we can't control that behaviour as far as I know.

Hope this helps!

**Comment by lakshya-bs** [comment]
Attaching a screen recording from Safari browser. In this hovering on items announces it as item added to selection and on removing hover from that item it says item removed from selection but thats not expected on hover. This should have been the case when I click the item.

Uploading Screen Recording 2023-08-10 at 6.48.43 PM.mov…

**Comment by Lakshya-Poddar** [comment]
Attaching a screen recording from Safari Browser. In this hovering on items announces it as `item added to selection` and on removing hover from that item it says `item removed from selection` this is not expected on hover but when I click on some item. 

https://github.com/tailwindlabs/headlessui/assets/43548762/1cab9a94-de7b-4d4f-9eed-32e7895a3169
